File: ProyectoCsharp/Modelo/RegistroIngreso.cs
using System;
using System.Collections.Generic;

namespace Modelo
{
    public class RegistroIngreso
    {
        private long id_Ingreso;
        private Proveedor proveedor;
        private AlmacenPrincipal almacen;
        private DateTime fecha;
        private DateTime fecha_proceso;
        private TTipoComprobante tipoComprobante;
        private double subtotal;
        private double igv;
        private double monto_igv;
        private double total;
        private TEstado estado;
        private List<DetalleRegistroIngreso> detalles;

        public RegistroIngreso()
        {
            this.Id_Ingreso = 0;
            this.proveedor = null;
            this.almacen = null;
            this.Subtotal = 0;
            this.Igv = 0;
            this.Monto_igv = 0;
            this.Total = 0;
            this.detalles = new List<DetalleRegistroIngreso>();
        }

        public RegistroIngreso(long id_Ingreso, Proveedor proveedor, AlmacenPrincipal almacen, DateTime fecha,
                    DateTime fecha_proceso, TTipoComprobante tipoComprobante, double subtotal,
                    double igv, double monto_igv, double total, TEstado estado)
        {

            this.Id_Ingreso = id_Ingreso;
            this.Proveedor = proveedor;
            this.Almacen = almacen;
            this.Fecha = fecha;
            this.Fecha_proceso = fecha_proceso;
            this.TipoComprobante = tipoComprobante;
            this.Subtotal = subtotal;
            this.Igv = igv;
            this.Monto_igv = monto_igv;
            this.Total = total;
            this.Estado = estado;
            this.detalles = new List<DetalleRegistroIngreso>();
        }

        //Destructor
        ~RegistroIngreso()
        {
            proveedor = null;
            almacen = null;
            detalles = null;
        }

        //Set y get
        public long Id_Ingreso { get => id_Ingreso; set => id_Ingreso = value; }
        public Proveedor Proveedor { get => proveedor; set => proveedor = value; }
        public AlmacenPrincipal Almacen { get => almacen; set => almacen = value; }
        public DateTime Fecha { get => fecha; set => fecha = value; }
        public DateTime Fecha_proceso { get => fecha_proceso; set => fecha_proceso = value; }
        public TTipoComprobante TipoComprobante { get => tipoComprobante; set => tipoComprobante = value; }
        public double Subtotal { get => subtotal; set => subtotal = value; }
        public double Igv { get => igv; set => igv = value; }
        public double Monto_igv { get => monto_igv; set => monto_igv = value; }
        public double Total { get => total; set => total = value; }
        public TEstado Estado { get => estado; set => estado = value; }

        public void AgregarDetalle(DetalleRegistroIngreso detalle)
        {
            detalles.Add(detalle);
        }
        public List<DetalleRegistroIngreso> GetDetalle()
        {
            return detalles;
        }

    }

}

File: TrabajoJavaLp2/src/modelo/RegistroIngreso.java
package modelo;
import java.util.Date;
import java.util.ArrayList;
public class RegistroIngreso{
	private long id_Ingreso;
	private Proveedor proveedor;
	private Almacen almacen;
	private Date fecha;
	private Date fecha_proceso;
	private TTipoComprobante tipoComprobante;
	private double subtotal;
	private double igv;
	private double monto_igv;
	private double total;
	private TEstado estado;
	private ArrayList<DetalleRegistroIngreso> detalles;
	

	public RegistroIngreso(){
		this.id_Ingreso = 0;
        this.proveedor=null;
        this.almacen=null;
        this.subtotal=0;
        this.igv=0;
        this.monto_igv=0;
        this.total=0;
        this.detalles = new ArrayList<DetalleRegistroIngreso>();
	}

	public RegistroIngreso(long id_Ingreso, Proveedor proveedor, Almacen almacen,Date fecha,
				Date fecha_proceso, TTipoComprobante tipoComprobante, double subtotal, 
				double igv, double monto_igv,double total, TEstado estado){
		
		this.id_Ingreso = id_Ingreso;
        this.proveedor=proveedor;
        this.almacen=almacen;
        this.fecha=fecha;
        this.fecha_proceso=fecha_proceso;
        this.tipoComprobante=tipoComprobante;
        this.subtotal=subtotal;
        this.igv=igv;
        this.monto_igv=monto_igv;
        this.total=total;
        this.estado=estado;
        this.detalles = new ArrayList<DetalleRegistroIngreso>();
	}

	public void setProveedor(Proveedor proveedor){
		this.proveedor = proveedor;
	}
	public Proveedor getProveedor(){
		return this.proveedor;
	}

	public void setId_ingreso(long id_Ingreso){
		this.id_Ingreso = id_Ingreso;
	}
	public long getId_ingreso(){
		return this.id_Ingreso;
	}

	public void setAlmacen(Almacen almacen){
		this.almacen = almacen;
	}
	public Almacen getAlmacen(){
		return this.almacen;
	}


	public void setFecha(Date fecha){
		this.fecha = fecha;
	}
	public Date getFecha(){
		return this.fecha;
	}


	public void setFecha_proceso(Date fecha_proceso){
		this.fecha_proceso = fecha_proceso;
	}
	public Date getFecha_proceso(){
		return this.fecha_proceso;
	}


	public void setTipoComprobante(TTipoComprobante tipoComprobante){
		this.tipoComprobante = tipoComprobante;
	}
	public TTipoComprobante getTipoComprobante(){
		return this.tipoComprobante;
	}

	public void setSubtotal(double subtotal){
		this.subtotal = subtotal;
	}
	public double getSubtotal(){
		return this.subtotal;
	}
	
	public void setIgv(double igv){
		this.igv = igv;
	}
	public double getIgv(){
		return this.igv;
	}
	
	public void setMonto_igv(double monto_igv){
		this.monto_igv = monto_igv;
	}
	public double getMonto_igv(){
		return this.monto_igv;
	}
	
	public void setTotal(double total){
		this.total = total;
	}
	public double getTotal(){
		return this.total;
	}

	public void setDetalle(DetalleRegistroIngreso detalle){
		detalles.add(detalle);
	}
	public ArrayList<DetalleRegistroIngreso> getDetalle(){
		return detalles;
	}

}